en/02_Developer_Guides/02_Controllers/02_Routing.md

```diff
@@ -303,6 +303,53 @@ class BreadAPIController extends Controller
+
+### How it works
+
+[`Director::handleRequest()`](api:SilverStripe\Control\Director::handleRequest()) begins the url parsing process by parsing the start of the URL and workng out which [`RequestHandler`](api:SilverStripe\Control\RequestHandler) subclass to use, looking in:
+
+- Routes set in yml config under Director `rules`
+- For `/admin` routes specifically, the `$url_segment` config for subclasses of `LeftAndMain`
+
+When a [`RequestHandler`](api:SilverStripe\Control\RequestHandler) matching the first portion of the URL is found, the `handleRequest()` method on the matched [`RequestHandler`](api:SilverStripe\Control\RequestHandler) subclass is called. This passes control to the matched [`RequestHandler`](api:SilverStripe\Control\RequestHandler) and the URL is effectively truncated from the left, removing the portion that lead to ths point.
+
+From there regular request handling occurs and URL will be check to see if it matches `$allowed_actions` on the [`RequestHandler`](api:SilverStripe\Control\RequestHandler), possibly routed via `$url_handlers`. If an `$allowed_action` i.e. method on the [`RequestHandler`](api:SilverStripe\Control\RequestHandler) is matched, if that method returns a [`RequestHandler`](api:SilverStripe\Control\RequestHandler), then control will now be passed to this nested [`RequestHandler`](api:SilverStripe\Control\RequestHandler) and the URL is again effectively truncated from the left.
+
+### Example of a nested RequestHandler being returned in an action method
+
+- There is yml routing directing the path `one` to `RequestHandlerOne`
+- There is a class `RequestHandlerOne` with `$allowed_actions = ['two']`, and a `two()` method returning `RequestHandlerTwo::create();`
+- There is a class `RequestHandlerTwo` with `$allowed_actions = ['hello']`, and a `hello()` method returning `HTTPResponse::create()->setBody('hello');`
+- Navigating to the URL `/one/two/hello` will return a response with a body of "hello"
+
+### Form, HasRequestHandler, FormRequestHandler, and FormField
+
+[`Form`](api:SilverStripe\Forms\Form) does not extend [`RequestHandler`](api:SilverStripe\Control\RequestHandler), instead it implements the [`HasRequestHandler`](api:SilverStripe\Control\HasRequestHandler) interface that requires a method `getRequestHandler()`. [`Form::getRequestHandler()`](api:SilverStripe\Forms\Form::getRequestHandler()) returns a [`FormRequestHandler`](api:SilverStripe\Forms\FormRequestHandler) which is a subclass of [`RequestHandler`](api:SilverStripe\Control\RequestHandler).
+
+Subclasses of [`RequestHandler`](api:SilverStripe\Control\RequestHandler) and implementors of [`HasRequestHandler`](api:SilverStripe\Control\HasRequestHandler) are treated the same because they will both end up calling `handleRequest()` on the subclass of the appropriate [`RequestHandler`](api:SilverStripe\Control\RequestHandler) subclass.
```

Comment:
```suggestion
Subclasses of `RequestHandler` and implementors of `HasRequestHandler` are treated the same because they will both end up calling `handleRequest()` on the subclass of the appropriate `RequestHandler` subclass.
```

Reply:
Updated